src/Sparrow/Json/JsonOperationContext.cs

```diff
@@ -35,9 +32,10 @@ public class JsonOperationContext : IDisposable
 
         private int _lastStreamSize = 4096;
 
-        public JsonOperationContext(UnmanagedBuffersPool pool)
+        public JsonOperationContext()
         {
-            Pool = pool;
+            _arenaAllocator = new ArenaMemoryAllocator();
+            _arenaAllocatorForLongLivedValues = new ArenaMemoryAllocator(16 * 1024);
```

Comment:
Because we are creating a memory allocator per context. First line of the constructor is 
```csharp
_ptrStart = _ptrCurrent = (byte*)Marshal.AllocHGlobal(initialSize).ToPointer();
```
Dispose also dealloc the memory. 

What strikes me as odd is that this is exactly what we are doing with ```ByteString```. The ArenaAllocator looks like a replication of the same functionality with an slightly bigger buffer.